src/neoxp/Node/NodeUtility.cs

```diff
@@ -138,5 +142,127 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
 
             return tx;
         }
+
+        public const byte Prefix_Contract = 8;
+        private const byte Prefix_NextAvailableId = 15;
+
+        private static int LastUsedContractId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.TryGet(key);
+            return (int)(BigInteger)item;
+        }
+
+        private static void SetLastUsedContractId(DataCache snapshot, int newId)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
```

Comment:
I think item needs to be incremented by one here. You're calling `SetLastUsedContractId` with the id of the contract downloaded from remote chain. Pretty sure the next contract deployed will overwrite the downloaded contract

Reply:
This is not needed because a contract deploy will add +1 to the last used id, see [deploy](https://github.com/neo-project/neo/blob/2a64c1cc809d1ff4b3a573c7c22bffbbf69a738b/src/neo/SmartContract/Native/ContractManagement.cs#L187) and [GetNextAvailableId](
https://github.com/neo-project/neo/blob/2a64c1cc809d1ff4b3a573c7c22bffbbf69a738b/src/neo/SmartContract/Native/ContractManagement.cs#L79-L83)

Reply:
I can see where the confusion might come from; `Prefix_NextAvailableId` is a poor name choice (I kept it the same as the `neo` source). It's actually more a `Prefix_LastUsedId`

Reply:
> I can see where the confusion might come from; `Prefix_NextAvailableId` is a poor name choice (I kept it the same as the `neo` source). It's actually more a `Prefix_LastUsedId`

I'm not sure that's right. My read of this is that the stored value is returned as the ID to use while storage is updated to increment by one. If the currently stored value is 10, GetNextAvailableId returns 10 and updates storage to be 11.

``` csharp
        private int GetNextAvailableId(DataCache snapshot)
        {
            StorageItem item = snapshot.GetAndChange(CreateStorageKey(Prefix_NextAvailableId));
            int value = (int)(BigInteger)item;
            item.Add(1);
            return value;
        }
```